src/toast/utils.py

```diff
@@ -831,3 +831,86 @@ def is_empty(self):
             if (len(v) > 0) and not all(x is None for x in v):
                 return False
         return True
+
+
+def flagged_noise_fill(data, flags, buffer, poly_order=1):
```

Comment:
I think we ultimately want a unit test for a frequently used utility function like this but I would not hold up merging the PR.

Reply:
Definitely agree.  Also the correct way to do this would be to iteratively fill all gaps simultaneously with a constrained noise realization.  That kind of technique was expensive the last time I implemented code for such things (circa 2005, written in C).  I should dig that up and see if it is fast enough now for this purpose.

Reply:
This function has been improved now, and a unit test is in place, along with a standalone operator that just runs the gap filling.